coalib/core/DependencyTracker.py

```diff
@@ -0,0 +1,162 @@
+from coalib.core.Graphs import traverse_graph
+
+
+class DependencyTracker:
+    """
+    A ``DependencyTracker`` allows to register and manage dependencies between
+    objects.
+
+    This class uses a directed graph to track relations.
+
+    Add a dependency relation between two objects:
+
+    >>> object1 = object()
+    >>> object2 = object()
+    >>> tracker = DependencyTracker()
+    >>> tracker.add(object2, object1)
+
+    This would define that ``object1`` is dependent on ``object2``.
+
+    If you define that ``object2`` has its dependency duty fulfilled, you can
+    resolve it:
+
+    >>> resolved = tracker.resolve(object2)
+    >>> resolved
+    {<object object at ...>}
```

Comment:
yes that's the `repr` of an object created by the built-in function `object()`. The class name of such an object is `object` ;)